### ISSUE TITLE
Question about  the rules for calculating FID indicators

### ISSUE THREAD

**Issue by gdut-cjy** [geyuying/PF-AFN#6]
Since the clothflow,this was the other job on appearance warping,awesome!

And how is your FID(10.09) calculated? The test dataset has 2032 clothing images, there are countless kinds of try-on possibilities.Maybe you have the file like the demo.txt you provided?

**Comment by geyuying** [comment]
Yes, we follow the previous work  to evaluate our models on VITON test set with the [https://github.com/minar09/cp-vton-plus/blob/master/data/test_pairs.txt](url) .

**Comment by gdut-cjy** [comment]
Hi，I use the pytorch-fid to test the result，but I can‘t get the experimental results as you. Maybe my method is wrong，could you tell me how to test the fid？
![image](https://user-images.githubusercontent.com/68991221/111055443-e241b680-84b0-11eb-94d2-f30626c17506.png)

**Comment by geyuying** [comment]
Hi, can you show some results from you "prediction" folder to see whether the testing is implemented correctly?

**Comment by gdut-cjy** [comment]
![4](https://user-images.githubusercontent.com/68991221/111056172-08b72000-84b8-11eb-8651-3d648c3dbb60.jpg)
![7](https://user-images.githubusercontent.com/68991221/111056173-094fb680-84b8-11eb-8e5b-1a46633e53a6.jpg)
![2](https://user-images.githubusercontent.com/68991221/111056174-0bb21080-84b8-11eb-894c-4e7f0a32b8a3.jpg)
![6](https://user-images.githubusercontent.com/68991221/111056176-0ce33d80-84b8-11eb-8f98-5d00f55ec86f.jpg)
![12](https://user-images.githubusercontent.com/68991221/111056177-0e146a80-84b8-11eb-97e1-f67e6668cb81.jpg)
![13](https://user-images.githubusercontent.com/68991221/111056179-0fde2e00-84b8-11eb-951e-483fb8735192.jpg)
![23](https://user-images.githubusercontent.com/68991221/111056180-12408800-84b8-11eb-8a15-211aead466ed.jpg)
![36](https://user-images.githubusercontent.com/68991221/111056183-14a2e200-84b8-11eb-9735-5adde10cdac3.jpg)
![38](https://user-images.githubusercontent.com/68991221/111056186-166ca580-84b8-11eb-95ef-c0ef09875d25.jpg)

**Comment by gdut-cjy** [comment]
![1](https://user-images.githubusercontent.com/68991221/111056193-24222b00-84b8-11eb-8802-526ac8e7a7bd.jpg)![3](https://user-images.githubusercontent.com/68991221/111056195-2ab0a280-84b8-11eb-8b85-f89a8ca90ba2.jpg)![21](https://user-images.githubusercontent.com/68991221/111056199-2edcc000-84b8-11eb-9163-530212308e48.jpg)![15](https://user-images.githubusercontent.com/68991221/111056202-313f1a00-84b8-11eb-835b-5b140aec1a14.jpg)![12](https://user-images.githubusercontent.com/68991221/111056203-3308dd80-84b8-11eb-8cac-e97dad1124b9.jpg)![38](https://user-images.githubusercontent.com/68991221/111056207-37cd9180-84b8-11eb-9477-227cf191e4e0.jpg)![2](https://user-images.githubusercontent.com/68991221/111056194-25ebee80-84b8-11eb-9805-4db1a3857289.jpg)![121](https://user-images.githubusercontent.com/68991221/111056221-559af680-84b8-11eb-8209-2130d8fcf0fd.jpg)

**Comment by geyuying** [comment]
![image](https://user-images.githubusercontent.com/25278412/111056371-93e4e580-84b9-11eb-84be-b2ca1978fe11.png)  ![image](https://user-images.githubusercontent.com/25278412/111056453-e0302580-84b9-11eb-872f-226c811dc4de.png)

The above are some of our results with the provided checkpoint. It seems that your testing environment is not the same as ours listed in README.  Please see the issue 3 in the closed issues to solve this problem.

**Comment by gdut-cjy** [comment]
okk,I will check the env,thanks!

**Comment by gdut-cjy** [comment]
After correcting the env, the effect get better!
![image](https://user-images.githubusercontent.com/68991221/111258256-9715d800-8657-11eb-8dce-f92cff371462.png)

**Comment by geyuying** [comment]
Hi, the FID you calculated is still worse than ours using the same checkpoints. We use https://github.com/mseitzer/pytorch-fid for evaluation.

**Comment by lyx0208** [comment]
I get FID:  9.503645410720281 with your pre-trained model, maybe this is because I use a higher version of PyTorch and other packages?

**Comment by geyuying** [comment]
@lyx0208 Smaller FID can indicate better results. Can you show some results on VITON test set? And which repo did you use to calculate the FID?

**Comment by lyx0208** [comment]
@geyuying I just use the link you post above to calculate FID.
I know smaller FID means better results, so it seems a little bit abnormal to me. Here are some results:
![1](https://user-images.githubusercontent.com/65213004/111426118-1cb88700-872f-11eb-9c88-9af86f4090c4.jpg)
![3](https://user-images.githubusercontent.com/65213004/111426141-23df9500-872f-11eb-9cdc-1d8f531e9a0b.jpg)
![4](https://user-images.githubusercontent.com/65213004/111426146-25a95880-872f-11eb-8d3e-1a277b5d0e38.jpg)
![19](https://user-images.githubusercontent.com/65213004/111426158-2b06a300-872f-11eb-9558-799b7a497d3f.jpg)

btw, when will be the codes related to training be released, I've send you an email this morning for some reasons related to this.

**Comment by geyuying** [comment]
@lyx0208  Hi, I did not recieve any email. Can you send it again? My email address is yyge13@gmail.com

**Comment by lyx0208** [comment]
@geyuying Sent again, maybe Gmail cannot receive emails from my school email address

**Comment by VVingerfly** [comment]
Hi, @lyx0208 @gdut-cjy

**Comment by VVingerfly** [comment]
Hi, Yuying @geyuying 
Thanks for your awesome work! The results are really amazing！

Could you tell me your enviroment when testing CP-VTON+?  I got a FID **10.186522** for PF-AFN, which is similar to yours 10.09. But for CP-VTON+, I got **31.151999** which is much higher then your result 21.08. Do you have any idea for this result?

Thanks a lot!
B.R.

**Comment by xxxxl888** [comment]
> After correcting the env, the effect get better! ![image](https://user-images.githubusercontent.com/68991221/111258256-9715d800-8657-11eb-8dce-f92cff371462.png)

Hello, I would like to ask how you calculate the FID score? Are you using the try-on results generated from the test and the images of the person inputted? My FID score is particularly high.
